### PR TITLE
Support multiple base images for ami builds

### PR DESCRIPTION
**Pull request by garyservin**
Add support to have multiple ubuntu distro versions for ami builds

### CODE REVIEW

File: tailor_image/create_image.py [hunk 1]
```diff
@@ -141,10 +142,25 @@ def create_image(name: str, distribution: str, apt_repo: str, release_track: str
 
     elif build_type == 'ami':
         image_name = f'{organization}_{name}_{distribution}_ami_{release_label}'
+        # Get ami-id for base image
+        ec2 = boto3.client('ec2')
+        source_image_name = recipe[name]['source_ami'].replace('$distribution', distribution)
+        source_ami_id = None
+        images = ec2.describe_images(Owners=['self'])['Images']
+        for image in images:
```
Comment (paulbovbel) [inline tailor_image/create_image.py:150]: this is a linear search through all our AMIs. is this going to get painful as our list of AMIs grow? Do we prune AMIs currently?

File: environment/image_recipes/ami/ami.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
     "aws_secret_key": "",
     "aws_region": "us-east-1",
 
-    "source_ami": "ami-0bfb5f753b6c2d30b",
+    "source_ami_id": "ami-0bfb5f753b6c2d30b",
```
Comment (paulbovbel) [inline environment/image_recipes/ami/ami.json:20]: what's this AMI id used for?

File: tailor_image/create_image.py (hunk 1 above)
Comment (paulbovbel) [reply tailor_image/create_image.py:150]: Is there a 'find' AMI type method?

File: tailor_image/create_image.py [hunk 3]
```diff
@@ -141,10 +142,25 @@ def create_image(name: str, distribution: str, apt_repo: str, release_track: str
 
     elif build_type == 'ami':
         image_name = f'{organization}_{name}_{distribution}_ami_{release_label}'
+        # Get ami-id for base image
+        ec2 = boto3.client('ec2')
+        source_image_name = recipe[name]['source_ami'].replace('$distribution', distribution)
+        source_ami_id = None
+        images = ec2.describe_images(Owners=['self'])['Images']
```
Comment (paulbovbel) [inline tailor_image/create_image.py:149]: Is it possible to filter by name here? Looks like there's a filter argument

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#EC2.Client.describe_images

File: tailor_image/create_image.py (hunk 1 above)
Comment (garyservin) [reply tailor_image/create_image.py:150]: we overwrite the sim images, so we only keep the latest for each release/feature atm

**Review by MJohnson459 — APPROVED** [review]

**Review by paulbovbel — APPROVED** [review]

**Comment by paulbovbel** [comment]
Can we add an AMI build to the toydistro for testing purposes?

**Pull request merged**